src/Sulu/Component/Content/Document/Subscriber/StructureRemoveSubscriber.php

```diff
@@ -57,19 +58,19 @@ public function handleRemove(RemoveEvent $event)
 
     public function removeDocument($document)
     {
-        // TODO: This is not a good indicator. There should be a RoutableBehavior here.
-        if (!$document instanceof StructureBehavior) {
-            return;
-        }
-
         if ($document instanceof ChildrenBehavior) {
             foreach ($document->getChildren() as $child) {
                 $this->removeDocument($child);
             }
         }
 
-        $this->removeReferences($document);
-        $this->recursivelyRemoveRoutes($document);
+        if ($document instanceof StructureBehavior) {
+            $this->removeReferences($document);
+        }
+
+        if ($document instanceof RoutableBehavior) {
```

Comment:
i would write a issue for that because its not really related to this PR and this wouldnt get even bigger and bigger